site/zenodo_rdm/migrator/transform/entries/records/records.py

```diff
@@ -44,18 +46,27 @@ def _recid(self, entry):
 
     def _pids(self, entry):
         record = entry["json"]
+
         r = {
             "oai": {
                 "provider": "oai",
                 "identifier": record["_oai"]["id"],
             },
         }
         if record.get("doi"):
-            r["doi"] = {
-                "client": "datacite",
-                "provider": "datacite",
-                "identifier": record["doi"],
-            }
+            _doi = record["doi"]
+            if _doi.startswith(ZENODO_DATACITE_PREFIX):
+                r["doi"] = {
+                    "client": "datacite",
+                    "provider": "datacite",
+                    "identifier": record["doi"],
+                }
```

Comment:
shelve: there are some [edge cases](https://github.com/zenodo/zenodo/blob/b4a4ab88238619811488777f6a2cc92b5d490a94/zenodo/modules/records/config.py#L115) that we need to take into account.

Reply:
logged it here https://github.com/zenodo/zenodo-rdm/issues/305